### PR TITLE
FISH-6578 added jna required dependency to packager and upgraded it to latest

### LINKED ISSUE #5581: Enhancement: Update JNA in order to work with Apple Silicon (aarm64e) architecture / FISH-6578
# Description #
----------

Update JNA (https://github.com/java-native-access/jna, `net.java.dev.jna:jna`) in order to work with Apple Silicon (aarm64e) architecture.

## Expected Outcome

`pom.xml`->`jna.version` used is 5.10.0
Payara Embedded contains jna in Version 5.10.0.

## Current Outcome

jna.version is 5.2.0 (and 4.5.2 for libpam4j)

## Context

I am using payara embedded (5.2020.6 and 5.2021.10) in Tests. It is a very simple solution to provide JakartaEE Spec implementations matching the server side implementation to unit and integration tests.
I am also using Vaadin Testbench which uses JNA for license validation. 
Now there are two JNA Versions in my test classpath: the one contained in payara-embedded (5.2.0) and the one provided by vaadin testbench (5.7.0). License validation fails on Apple Silicon (aarm64e) if V5.2.0 is used (see https://github.com/vaadin/testbench/issues/1308).

5.2.0 predates Apple Silicon (Release Dec. 2018)

### PR DESCRIPTION
**Pull request by lprimak**
## Description

Pam4j is missing the required JNA dependency in the packager 
Latest JNA so it runs on apple silicon.
Only other possibility is to put JNA in modules/ putting it in lib/ doesn’t work 

### Testing Performed

fixes #5581 

### Testing Environment

Mac, Linux

### CODE REVIEW

**Comment by lprimak** [comment]
Jenkins test please

**Comment by lprimak** [comment]
jenkins test

**Comment by JamesHillyard** [comment]
Hi @lprimak,

This seems to cause an issue with the build when using the `BuildExtras` profile:

```[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  06:54 min (Wall Clock)
[INFO] Finished at: 2022-10-25T09:44:40+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jvnet.maven-antrun-extended-plugin:maven-antrun-extended-plugin:1.43:run (default) on project payara-embedded-web: An Ant BuildException has occured: The following error occurred while executing this line:
[ERROR] C:\Users\JamesHillyard\IdeaProjects\Payara\appserver\extras\embedded\build.xml:73: The following error occurred while executing this line:
[ERROR] C:\Users\JamesHillyard\IdeaProjects\Payara\appserver\extras\embedded\build.xml:129: java.lang.RuntimeException: data starting at 9 is in unknown format
[ERROR] -> [Help 1]
[ERROR] Failed to execute goal org.jvnet.maven-antrun-extended-plugin:maven-antrun-extended-plugin:1.43:run (default) on project payara-embedded-all: An Ant BuildException has occured: The following error occurred while executing this line:
[ERROR] C:\Users\JamesHillyard\IdeaProjects\Payara\appserver\extras\embedded\build.xml:73: The following error occurred while executing this line:
[ERROR] C:\Users\JamesHillyard\IdeaProjects\Payara\appserver\extras\embedded\build.xml:129: java.lang.RuntimeException: data starting at 9 is in unknown format
[ERROR] -> [Help 1]```

**Comment by cubastanley** [comment]
Jenkins test please

**Comment by lprimak** [comment]
Jenkins test

**Comment by lprimak** [comment]
Jenkins test

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
Jenkins test

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
Jenkins test

**Comment by lprimak** [comment]
Please merge me :)

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
Jenkins test

**Comment by lprimak** [comment]
@JamesHillyard please merge me :)

**Comment by JamesHillyard** [comment]
Hi @lprimak,

My apologies for the delay, I have attempted to review these changes yesterday, although it appears there is a failing test caused by these changes. This was found by running `mvn clean install -PQuickBuild -T 10`
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.089 s <<< FAILURE! - in fish.payara.server.internal.extras.embedded.tests.jersey.JerseyClasspathTest
[ERROR] fish.payara.server.internal.extras.embedded.tests.jersey.JerseyClasspathTest.jerseyShouldInitialize  Time elapsed: 1.03 s  <<< ERROR!
java.lang.IllegalStateException: InjectionManagerFactory not found.
        at org.glassfish.jersey.internal.inject.Injections.lambda$lookupInjectionManagerFactory$0(Injections.java:74)
        at java.util.Optional.orElseThrow(Optional.java:290)
        at org.glassfish.jersey.internal.inject.Injections.lookupInjectionManagerFactory(Injections.java:74)
        at org.glassfish.jersey.internal.inject.Injections.createInjectionManager(Injections.java:44)
        at org.glassfish.jersey.client.ClientConfig$State.initRuntime(ClientConfig.java:413)
        at org.glassfish.jersey.internal.util.collection.Values$LazyValueImpl.get(Values.java:317)
        at org.glassfish.jersey.client.ClientConfig.getRuntime(ClientConfig.java:820)
        at org.glassfish.jersey.client.ClientRequest.getClientRuntime(ClientRequest.java:176)
        at org.glassfish.jersey.client.ClientRequest.getInjectionManager(ClientRequest.java:567)
        at org.glassfish.jersey.client.JerseyWebTarget.onBuilder(JerseyWebTarget.java:371)
        at org.glassfish.jersey.client.JerseyWebTarget.request(JerseyWebTarget.java:199)
        at org.glassfish.jersey.client.JerseyWebTarget.request(JerseyWebTarget.java:38)
        at fish.payara.server.internal.extras.embedded.tests.jersey.JerseyClasspathTest.jerseyShouldInitialize(JerseyClasspathTest.java:69)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
        at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
        at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:364)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:237)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:158)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
        at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)

[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR]   JerseyClasspathTest.jerseyShouldInitialize:69 » IllegalState InjectionManagerF...
[INFO]
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
```

Could you please address this failing test, and then everything looks good and I will be glad to get this merged ASAP.

Thanks,
James

**Comment by lprimak** [comment]
Hi @JamesHillyard Can you see how this is at all related to the fix?
I really can't... :)
It clearly passes the CI with the same tests and works on my machine as well.
Thanks for looking!

**Comment by JamesHillyard** [comment]
Hi @lprimak,

I appear to have copied the wrong command previously, this module is only built when using the BuildExtras profile. Could you please test again with `mvn clean install -PBuildExtras -pl :embedded-vs-jersey`

I thought the same though, and so asked internally for someone else to try compile this using `mvn clean install -PBuildExtras -T 10` but they got the same failure on Linux ruling out my environment. Jenkins doesn't use the BuildExtras profile and so will not be executing this test

Thanks,
James

**Comment by lprimak** [comment]
Can you please try to build the same master without my changes?  I can’t reproduce it.

**Comment by lprimak** [comment]
NVM, I was able to reproduce, still have no idea what could cause this

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
@cubastanley I think you broke CI :)

**Comment by lprimak** [comment]
Ok @JamesHillyard I figured it out. There was a functionality change in the antrun plugin and the extended plugin wasn't doing the right things. 
Please see https://github.com/payara/patched-src-maven-antrun-extended-plugin/pull/1
and, change the version to 1.45 and deploy it into Payara nexus.
Then, this PR should compile and work.

Thanks for all your efforts to get this through. I didn't think in a million years this would be that difficult.

File: nucleus/pom.xml [hunk 1]
```diff
@@ -361,7 +361,7 @@
                 <plugin>
                     <groupId>org.jvnet.maven-antrun-extended-plugin</groupId>
                     <artifactId>maven-antrun-extended-plugin</artifactId>
-                    <version>1.43</version>
+                    <version>1.45</version>
```
Comment (JamesHillyard) [inline nucleus/pom.xml:364]:
```suggestion
                    <version>1.43.payara-p1</version>
```

Comment (JamesHillyard) [reply nucleus/pom.xml:364]: @lprimak Same again to update this PR - Then this can be run through Jenkins

**Comment by lprimak** [comment]
Jenkins test

**Review by JamesHillyard — APPROVED** [review]

**Comment by JamesHillyard** [comment]
Thank you for your contribution and all the effort in getting the tests to pass, it's much appreciated. I will port the changes into Payara Enterprise and Payara 6 shortly

**Pull request merged**